### ISSUE TITLE
authenticating against index.docker.io

### ISSUE THREAD

**Issue by hone** [camallo/dkregistry-rs#178]
I'm trying to pull a docker manifest from the public registry, but I'm getting a 401. There's a way to get a Bearer Token from auth.docker.io, but I don't know how pass that bearer token into the Client. With the `auth` module being private, it seems not possible to do this without a username/password.

**Comment by hone** [comment]
As a comparison the workflow I'm looking for is similar to the other docker registry crate [oci-registry-client](https://github.com/ecarrara/oci-registry-client) as described in the `README`:

```
    let mut client = DockerRegistryClientV2::new(
        "registry.docker.io",
        "https://registry-1.docker.io",
        "https://auth.docker.io/token"
    );
    let token = client.auth("repository", "library/ubuntu", "latest").await?;
    client.set_auth_token(Some(token));

    let manifest = client.manifest("library/ubuntu", "latest").await?;
```

**Comment by acim** [comment]
Auth from this example works for me: https://github.com/camallo/dkregistry-rs/blob/master/examples/image.rs

**Comment by steveeJ** [comment]
IIRC this has been solved in #168. Please reopen this if it's still an issue.

**Comment by hone** [comment]
Thanks!
